src/update-feeds/update-feeds.ts

```diff
@@ -162,7 +164,154 @@ export const updateFeeds = async (_batch: ReturnType<typeof getFeedsToUpdate>, _
   // batch, execute
 };
 
-export const processBatch = async (batch: ReadDapiWithIndexResponse[], providerName: Provider, chainId: string) => {
+export const deepCheckFeeds = async (
```

Comment:
You should modify the `getFeedsToUpdate` because it needs to determine which beacons are to be updated. This information should be propagated to `updateFeeds` function which should only update those beacons.